Fix EZP-24040: Date/Time attributes are not timezone friendly

yannickroger opened the pull request:
Link: https://jira.ez.no/browse/EZP-24040

## Description
This PR contains two fixes for Date and Time attributes
### Time
The time is now stored and displayed in UTC. It doesn't have to be timezone aware are it's just a plain data.

### DateAndTime
The API stores DateAndTime fields in UTC format and doesn't store the timezone. This patch converts back and forth UTC to local time in order to display something meaningful to the user.

## Test
(Lots of) manual test. Updated Unit test. It is at the moment not possible to automate more testing because of locale limitation in the headless browser used to run our unit tests.

## TODO
- [x] Cleanup

dpobel commented on Resources/public/js/views/fields/ez-dateandtime-editview.js:
```diff
@@ -264,6 +264,19 @@ YUI.add('ez-dateandtime-editview', function (Y) {
+         *
+         * @protected
+         * @method _getUtcTimeStamp
+         * @param {Integer} localizedTimestamp
```
in JS, it's `{Number}` not `{Integer}`

dpobel commented on Resources/public/js/views/fields/ez-dateandtime-editview.js:
```diff
@@ -264,6 +264,19 @@ YUI.add('ez-dateandtime-editview', function (Y) {
+         * @protected
+         * @method _getUtcTimeStamp
+         * @param {Integer} localizedTimestamp
+         * @return {Integer} Timestamp converted to UTC
```
you should mention that it's in millisecond.

dpobel commented on Resources/public/js/views/fields/ez-time-editview.js:
```diff
@@ -101,14 +101,17 @@ YUI.add('ez-time-editview', function (Y) {
+                    options = {hour: 'numeric', minute: 'numeric', second: 'numeric', timeZone:'UTC'};
                 }
+
+                time = new Date(field.fieldValue * 1000).toLocaleTimeString(undefined, options);
```
unfortunately, the support for the parameters of `toLocaleTimeString` is rather low (Safari does not support that it seems plus several mobile agent), so we cannot use it

dpobel commented:
+1

dpobel commented:
(The tests are are  failing on Travis, it seems like it's because of a missing dependency)

ping @mhyndle @andrerom

mhyndle commented:
+1

yannickroger commented:
@dpobel @mhyndle I fixed the unit test and while at it I added an extra commit fixing the error running `grunt test` because of locale missmatch.
Is it still ok for you guys ?

dpobel commented on Tests/js/views/fields/ez-dateandtime-view.html:
```diff
@@ -29,11 +29,11 @@
             },
             "ez-fieldview": {
-                requires: ['ez-templatebasedview'],
+                requires: ['ez-templatebasedview', 'datatype-date-format'],
```
not needed :)

dpobel commented:
besides the last inline comment, +1 (if Travis agrees :p)

mhyndle commented:
+1 confirmed
